webrtc.html

```diff
@@ -2389,6 +2389,88 @@
         </dd>
+
+        <dt>Promise&lt;void&gt; replaceTrack (MediaStreamTrack withTrack)</dt>
+
+        <dd>
+            <p>Attempts to replace the track being sent with another track
+            provided, without renegotiation.</p>
+
+            <p>When the <dfn id="dom-rtpsender-replacetrack"><code title=
+            "">replaceTrack()</code></dfn> method is invoked, the user agent MUST
+            run the following steps:</p>
+
+            <ol>
+              <li>
+                <p>Let <var>p</var> be a new promise.</p>
+              </li>
+
+              <li>
+                <p>Let <var>withTrack</var> be the argument to this method.</p>
+              </li>
+
+              <li>
+                <p>Run the following steps asynchronously:</p>
+              </li>
+              <ol>
+                <li>
+                  <p>If <var>withTrack's</var> <code><a href=
+                    "getusermedia.html#dom-mediastreamtrack-kind">kind</a></code>
+                  attribute differs from the
+                  </var> <code><a href=
+                    "getusermedia.html#dom-mediastreamtrack-kind">kind</a></code>
+                  of
+                  <code><a href="#dom-rtpsender-track">RTCRtpSender.track</a></code>,
+                  then reject
+                  <var>p</var> with <code>IncompatibleMediaStreamTrackError</code>
+                  and abort these steps.</p>
+                </li>
+                <li>
+                  Set the
+                  <code><a href="#dom-rtpsender-track">RTCRtpSender.track</a></code>
+                  attribute to <var>withTrack</var>.
+                <li>
+                  <p>Determine if negotiation is needed to transmit
+                  <var>withTrack</var> in place of the sender's existing track.
+                  Ignore what <code>MediaStream</code> the track resides in and the
+                  track's id in this determination.
+                </li>
+                <ul>
+                  <li>
+                    If no negotiation is needed, then have the sender seamlessly
+                    switch to transmitting <var>withTrack</var> in place of what
+                    it is sending, without negotiating. On the remote receiving
+                    end, the track maintains its existing grouping and id until
+                    the connection ends or a renegotiation happens, whichever is
+                    sooner. If a renegotiation happens, then the remote receiver
+                    SHOULD use SSRCs to detect tracks that have been renamed in
+                    this manner, and update their ids and groupings, rather than
+                    treat them as new tracks.</p>
+                  </li>
+                  <li>
+                    <p>If negotiation is needed, then fire a <a href=
+                      "#event-negotiation">negotiationneeded</a> event at the
+                    <a>RTCPeerConnection</a> associated with this sender.</p>
```

Comment:
As I understand the [session negotiation model](http://w3c.github.io/webrtc-pc/#session-negotiation-model), local settings still reflect what has been set locally, even when a PeerConnection becomes "dirty" with the `negotiationneeded` flag.

E.g. if I do `var newSender = pc.addTrack(newTrack);` then `newSender` will be in pc.getSenders() even if negotiation is pending. Since senders are extensions of PeerConnection, I would expect them to work the same, e.g. sender.track reflects what was last set.

If someone could validate my understanding then that would be great.